backend/sirius-web-spring-collaborative-diagrams/src/main/java/org/eclipse/sirius/web/spring/collaborative/diagrams/DiagramEventProcessor.java

```diff
@@ -94,7 +95,7 @@ public ISubscriptionManager getSubscriptionManager() {
                 IDiagramEventHandler diagramEventHandler = optionalDiagramEventHandler.get();
                 EventHandlerResponse eventHandlerResponse = diagramEventHandler.handle(this.editingContext, this.diagramContext, diagramInput);
                 if (eventHandlerResponse.getShouldRefreshPredicate().test(this.diagramContext.getDiagram())) {
-                    this.refresh();
+                    this.doRefresh(Optional.of(new NodeBoundsProvider()));
```

Comment:
The new providers NodeSizeProvider and NodePositionProvider are to be created in the sirius-web-diagram project as defined in the ADR. Unless I missed something, there is no need to change the API of the DiagramCreationService#refresh nor to change the DiagramEventProcessor.

Reply:
I anticipate the way we could provide the location of the node following a tool execution (by providing the location where the tool has been used). This information will be retrieve from the input passed to the eventProcessor and should be communicate to the DiagramCreationSerivce ?

Reply:
This information (some position) can be given to the DiagramCreationService to be ultimately given to the NodePositionProvider yes.